src/Native/System.Security.Cryptography.Native/pal_ecc_import_export.cpp

```diff
@@ -0,0 +1,520 @@
+    {
+        *d = const_cast<BIGNUM*>(EC_KEY_get0_private_key(key));
+        *cbD = BN_num_bytes(*d);
+    }
+    else
+    {
+        if (d)
+            *d = nullptr;
+
+        if (cbD)
+            *cbD = 0;
+    }
+
+    // success
+    return 1;
+
+error:
+    *cbQx = *cbQy = 0;
+    *qx = *qy = 0;
+    if (d) *d = nullptr;
+    if (cbD) *cbD = 0;
+    if (xBn) BN_free(xBn);
+    if (yBn) BN_free(yBn);
+    return rc;
+}
+
+extern "C" int32_t CryptoNative_GetECCurveParameters(
+    const EC_KEY* key,
+    int32_t includePrivate,
+    ECCurveType* curveType,
+    BIGNUM** qx, int32_t* cbQx,
+    BIGNUM** qy, int32_t* cbQy,
+    BIGNUM** d, int32_t* cbD,
+    BIGNUM** p, int32_t* cbP,
+    BIGNUM** a, int32_t* cbA,
+    BIGNUM** b, int32_t* cbB,
+    BIGNUM** gx, int32_t* cbGx,
+    BIGNUM** gy, int32_t* cbGy,
+    BIGNUM** order, int32_t* cbOrder,
+    BIGNUM** cofactor, int32_t* cbCofactor,
+    BIGNUM** seed, int32_t* cbSeed)
+{
+    // Get the public key parameters first in case any of its 'out' parameters are not initialized
+    int32_t rc = CryptoNative_GetECKeyParameters(key, includePrivate, qx, cbQx, qy, cbQy, d, cbD);
+
+    // Verify the out parameters. Note out parameters used to minimize pinvoke calls.
+    if (!p || !cbP ||
+        !a || !cbA ||
+        !b || !cbB ||
+        !gx || !cbGx ||
+        !gy || !cbGy ||
+        !order || !cbOrder ||
+        !cofactor || !cbCofactor ||
+        !seed || !cbSeed)
+    {
+        assert(false);
+
+        // Since these parameters are 'out' parameters in managed code, ensure they are initialized
+        if (p) *p = nullptr; if (cbP) *cbP = 0;
+        if (a) *a = nullptr; if (cbA) *cbA = 0;
+        if (b) *b = nullptr; if (cbB) *cbB = 0;
+        if (gx) *gx = nullptr; if (cbGx) *cbGx = 0;
+        if (gy) *gy = nullptr; if (cbGy) *cbGy = 0;
+        if (order) *order = nullptr; if (cbOrder) *cbOrder = 0;
+        if (cofactor) *cofactor = nullptr; if (cbCofactor) *cbCofactor = 0;
+        if (seed) *seed = nullptr; if (cbSeed) *cbSeed = 0;
+
+        return 0;
+    }
+
+    EC_GROUP* group = nullptr;
+    EC_POINT* G = nullptr;
+    EC_METHOD* curveMethod = nullptr;
+    BIGNUM* xBn = nullptr;
+    BIGNUM* yBn = nullptr;
+    BIGNUM* pBn = nullptr;
+    BIGNUM* aBn = nullptr;
+    BIGNUM* bBn = nullptr;
+    BIGNUM* orderBn = nullptr;
+    BIGNUM* cofactorBn = nullptr;
+    BIGNUM* seedBn = nullptr;
+
+    // Exit if CryptoNative_GetECKeyParameters failed
+    if (rc != 1) 
+        goto error;
+
+    xBn = BN_new();
+    yBn = BN_new();
+    pBn = BN_new();
+    aBn = BN_new();
+    bBn = BN_new();
+    orderBn = BN_new();
+    cofactorBn = BN_new();
+
+    if (!xBn || !yBn || !pBn || !aBn || !bBn || !orderBn || !cofactorBn)
+        goto error;
+
+    group = const_cast<EC_GROUP*>(EC_KEY_get0_group(key)); // curve
+    if (!group) 
+        goto error;
+
+    curveMethod = const_cast<EC_METHOD*>(EC_GROUP_method_of(group));
+    if (curveMethod == nullptr) 
```

Comment:
Nit: above we just do ```!group```, and here we do ```curveMethod == nullptr```... it'd be good to be consistent, at least within the same method :)